src/components/portfolio/PortfolioCard.tsx

```diff
@@ -0,0 +1,350 @@
+  ${tw`flex flex-col`}
+  gap: 8px;
+`;
+
+const InvestedType = styled.div`
+  display: flex;
+  align-items: center;
+  padding-left: 24px;
+  position: relative;
+  &::before {
+    content: '';
+    position: absolute;
+    left: 0px;
+    top: calc(50% - 4px);
+    width: 8px;
+    height: 8px;
+    border-radius: 100%;
+    background: ${TOKEN_PAIR_FIGURE_COLOR};
+  }
+  &:first-child::after {
+    content: '';
+    position: absolute;
+    left: 3px;
+    top: 16px;
+    width: 2px;
+    height: 24px;
+    background: ${TOKEN_PAIR_FIGURE_COLOR};
+  }
+`;
+
+const DashedDivider = styled.div`
+  margin-left: 8px;
+  margin-right: 8px;
+  position: relative;
+  flex-grow: 1;
+  &::before {
+    content: '';
+    position: absolute;
+    left: 0;
+    top: calc(50% - 1px);
+    width: 100%;
+    height: 1px;
+    border-bottom: 1px dashed ${DASHED_DIVIDER_BORDER_COLOR};
+  }
+`;
+
+const BodyDivider = styled.div`
+  width: 1px;
+  height: 88px;
+  background-color: ${BODY_DIVIDER_BG_COLOR};
+  @media (max-width: ${RESPONSIVE_BREAKPOINT_SMALL}) {
+    width: 100%;
+    height: 1px;
+  }
+`;
+
+export const ValueText = styled.span`
+  font-size: 32px;
+`;
+
+const PercentChangeContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 4px;
+  height: 28px;
+  border-radius: 8px;
+  padding: 6px;
+  font-size: 12px;
+  &:after {
+    content: '';
+    width: 14px;
+    height: 14px;
+    background-size: contain;
+    background-repeat: no-repeat;
+    background-position: center;
+  }
+`;
+
+export const PositivePercentChangeContainer = styled(PercentChangeContainer)`
+  background: ${POSITIVE_PERCENT_BG_COLOR};
+  color: ${POSITIVE_PERCENT_TEXT_COLOR};
+  &:after {
+    background-image: url(${PositiveChangeIcon});
+  }
+`;
+
+export const NegativePercentChangeContainer = styled(PercentChangeContainer)`
+  background: ${NEGATIVE_PERCENT_BG_COLOR};
+  color: ${NEGATIVE_PERCENT_TEXT_COLOR};
+  &:after {
+    background-image: url(${NegativeChangeIcon});
+  }
+`;
+
+// My simple solution to determining the prominent color of the icon
+// (Uses external library)
+// Feel free to replace this with something more robust if need be
+const getProminentColor = async (path: string) => {
+  const values = await prominent(path, { amount: 1 });
+  return `${values[0]}, ${values[1]}, ${values[2]}`;
+};
+
+const rgb = (color: string) => {
+  return `rgb(${color})`;
+};
+
+const rgba = (color: string, alpha: number) => {
+  return `rgba(${color}, ${alpha})`;
+};
+
+// Very naive logic to get the brighter color used to pick which color to
+// use for the shadow
+const getBrighterColor = (color0: string, color1: string) => {
+  const values0 = color0.split(',').map((v) => parseInt(v.trim()));
+  const values1 = color1.split(',').map((v) => parseInt(v.trim()));
+  const avg0 = (values0[0] + values0[1] + values0[2]) / 3;
+  const avg1 = (values1[0] + values1[1] + values1[2]) / 3;
+  return avg0 > avg1 ? color0 : color1;
+};
```

Comment:
Let's refactor these methods to be defined in a helper file and imported here, so they can be reused better for other components.